### PR TITLE
WRN-20224: [React18] Update templates

### PR DESCRIPTION
**Pull request by hong6316**
In this PR, updated react version to 18 and related packages.

### CODE REVIEW

File: packages/sandstone/template/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "@enact/ui": "^4.0.0",
     "ilib": "^14.6.0",
     "prop-types": "^15.6.2",
```
Comment (MikyungKim) [inline packages/sandstone/template/package.json]: Would you please update other dependencies to up to date as well?

Comment (MikyungKim) [reply packages/sandstone/template/package.json]: including enact packages

Comment (hong6316) [reply packages/sandstone/template/package.json]: OK

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**